indexer/editable_map_object.cpp

```diff
@@ -280,6 +280,14 @@ void EditableMapObject::SetWebsite(string website)
 void EditableMapObject::SetInternet(Internet internet)
 {
   m_metadata.Set(feature::Metadata::FMD_INTERNET, DebugPrint(internet));
+
+  if (internet == Internet::Unknown)
+  {
+    static const uint32_t wifiType = classif().GetTypeByPath({"internet_access", "wlan"});
```

Comment:
Ты меня правильно понял. Я именно имел ввиду что инициализация статика
синхронная, все остальное не синхронное. Но это означает что все равно
нужна проверка в какой то форме что статик уже проинициализирован. Я не
хочу сказать что это быстро или долго, я хочу сказать что здесь это не
нужно. Если есть желание что то по нормальному ускорить, я могу подкинуть
пару идей где и что в проекте стоит ускорять. Здесь же имхо простота и
читаемость лучше каких то premature optimizations.

18 авг. 2016 г. 20:12 пользователь "Arsentiy Milchakov" <
notifications@github.com> написал:

> In indexer/editable_map_object.cpp
> <https://github.com/mapsme/omim/pull/4080#discussion_r75348740>:
>
> > @@ -280,6 +280,14 @@ void EditableMapObject::SetWebsite(string website)
> >  void EditableMapObject::SetInternet(Internet internet)
> >  {
> >    m_metadata.Set(feature::Metadata::FMD_INTERNET, DebugPrint(internet));
> > +
> > +  if (internet == Internet::Unknown)
> > +  {
> > +    static const uint32_t wifiType = classif().GetTypeByPath({"internet_access", "wlan"});
>
> Если я правильно тебя понял, то ты хочешь сказать, что при использовании
> статической переменной присутствует оверхед, связанный с наличием
> блокировки для осуществления корректной работы в многопоточном окружении.
> Насколько мне известно, для статиков гарантируется только корректная
> инициализация в многопоточном окружении (пункт 6.7 стандарта). Ни о каких
> других гарантиях работы статиков в многопоточном окружении я упоминания не
> нашел и ранее не слышал. Чтобы не спорить впустую, я набросал пример
> использования статика с замером времени выполнения
> http://ideone.com/gVA8dD. В этом примере кусок кода со статической
> переменной работает в 100 раз быстрее. Если я не правильно тебя понял, то
> уточни пожалуйста, что ты имеешь ввиду.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/mapsme/omim/pull/4080/files/b280e24da30bd32269022090acdf4c51a93f8733#r75348740>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAbvp-ayJ8KXcEcNnA7-KlvAPiEGWAvjks5qhJKDgaJpZM4JmdzA>
> .
>